pkg/network/public/public.go

```diff
@@ -349,6 +356,15 @@ func EnsurePublicSetup(nodeID pkg.Identifier, inf *pkg.PublicConfig) (*netlink.B
 	return br, netlink.LinkSetUp(br)
 }
 
+func ensureTestNamespace(publicBrdige *netlink.Bridge) error {
```

Comment:
I don't see where u do `ensure` here. You just blindly create the namespace and the macvlan. This code will fail when it tries to run again or breaks the setup. 

The point of an ensure function is to validate and build the setup if not there. Hence you need to be check (validate) how good the setup is, and then create, or change it to match the requires setup.

This function can be called multiple times during the runtime for example because of a system update